### PR TITLE
Adds recoil damage to hulk('s arms) when smashing walls or machines

### PR DESCRIPTION
**Pull request by zxaber**
*Note that the original version of this PR was a wall smash removal, and so the comments mostly reflect that proposed change.*

## About The Pull Request
Adds recoil damage to hulks when they smash walls (and a bit for when smashing machines):

- Smashing a normal wall (and most other walls) will deal 20 damage to the arm used to smash it.
- Smashing an R-wall, Diamond wall, or Plastitanium wall will deal 41 damage to the arm.
- Attacking a machine will deal one tenth the machine's `damage_deflection` var, which is being used as a rough approximation of the structural hardness/toughness of the machine.
- After an attack, the hulk code evaluates the damage on the arm and applies a bone wound to the arm based on the arm's brute damage. Values are 35 to 41,  41 to 45, and 45 to 50 for low, medium, and high severity wounds, respectively. 

As per the above values, you can break up to three regular walls before your arm becomes unusable (this is per-arm), or two R-walls. Smashing airlocks also has a health cost, but you won't even get the first wound until you're half-way through breaking the sixth one.

To be clear, you only take damage from attacking a wall if the wall was actually broken by that attack. On the flipside, every attack against a machine will deal recoil damage. If turfs every get a proper health/integrity value, this can be changed.

## Why It's Good For The Game
Hulks getting to just delete any wall they feel like means that if and when hulk is released to the crew, the round might as well be over. There's no meaningful resource for distributing hulk, and no resource for smashing either, so it's pretty much a grief tool for the sake of it.

As per Cobby's request, I have changed this removal PR to deal damage to hulk arms instead. Hulks can still smash through a few walls if they really need to (though they're encouraged by the damage to use doorways), but not for free. Smashing windows, grills, or people does not have a health cost attached.

Machine recoil damage somewhat of a consistency thing (they're made of metal), but smashing machines just because you can should also be somewhat discouraged. The damage isn't very high comparatively; you can get through seven airlocks with one arm and still have the arm in a functional state.

## Changelog
:cl:
balance: Hulks now take damage to their arms for smashing walls and machines. You won't get through very many walls before you have to visit medical, though the damage for punching an airlock is quite a bit less.
/:cl:

### CODE REVIEW

**Comment by SkeletalElite** [comment]
Make it so they damage walls and it takes a while to beat through them instead, but have it still be possible.

**Comment by zxaber** [comment]
Walls don't have a health value. They're either broken or they're not.

**Comment by Rohesie** [comment]
Adding turf integrity wouldn't be a bad thing.

**Comment by Kelenius** [comment]
> Hulks getting to just delete any wall they feel like means that if and when hulk is released to the crew, the round might as well be over. There's no meaningful resource for distributing hulk, and no resource for smashing either, so it's pretty much a grief tool for the sake of it.

So is AA. With this nerf, hulk is useless. Stun immunity and punching people harder doesn't cut it by itself. Hulk is a powerful tool against malf AI and similar situations.

**Comment by ATH1909** [comment]
> > Hulks getting to just delete any wall they feel like means that if and when hulk is released to the crew, the round might as well be over. There's no meaningful resource for distributing hulk, and no resource for smashing either, so it's pretty much a grief tool for the sake of it.
> 
> So is AA. With this nerf, hulk is useless. Stun immunity and punching people harder doesn't cut it by itself. Hulk is a powerful tool against malf AI and similar situations.

Aye, this removes some of the utility of hulk, making it an entirely combat-focused mutation.

**Comment by ATH1909** [comment]
If people are bored enough to try to smash every wall on the station as a hulk, I think the round's gotten to a point where it should be ending soon anyway.

**Comment by HomocidlePancakes** [comment]
Pro Tip: Use Bolas

**Comment by zxaber** [comment]
AA grants access, it doesn't let you turn the station into one large room.

It's not (or shouldn't be) up to a few random assistants with hulk to decide the round is over.

I have not removed glass breaking or door smashing. If you want to hulk your way into a room, it's still available. This just removes people using hulk to systematically remove walls for no other reason except that they can.

**Comment by CPTANT** [comment]
This is like the only thing hulk still has.

**Comment by TheSilverNuke** [comment]
I 100% agree, but make it so that it can only not punch through rwalls. Normal walls should still be able to be broken, this will be a nice compromise.

**Comment by Kelenius** [comment]
> AA grants access, it doesn't let you turn the station into one large room.

You've never seen people giving AA to each other until half the station has it?

**Comment by Coastfront** [comment]
Just remove its ability to slap its way through R-Walls, but keep its ability to kool-aid man through non r-walls. It would still keep some utility while stopping hulk from just being able to break into wherever they want on the station.

**Comment by wesoda25** [comment]
Gene ball nerf

**Comment by Rohesie** [comment]
I'm in favor of this nerf.
If no maintainers disagree or make change requests in 72 hours from the opening of the PR I'm merging it.

**Comment by PKPenguin321** [comment]
@Rohesie An alternative that adds turf integrity instead of walls breaking randomly would be acceptable?

**Comment by Rohesie** [comment]
That would be preferred even, @PKPenguin321

**Comment by Qustinnus** [comment]
good luck getting someone to add turf integrity that isn't shit.

pro-tip: if you just add an integrity var to the turf thats shit.

**Comment by Rohesie** [comment]
Improvements can be done incrementally.

**Comment by Qustinnus** [comment]
not an improvement if its a completely separate system.

**Comment by Rohesie** [comment]
Mhm, but you can move `var/obj_integrity` down to the `/atom` level and extend the implementation for turfs, while keeping mobs and the rest out of it (redefine the procs at the mob level or whatever).
Since it's just two variables that won't be used and a bunch of procs it's not a noticeable cost and the system is ready to be expanded whenever.

**Comment by Qustinnus** [comment]
I'm saying that people most commonly shoot to "lets just make a var on /turf". which is shit. what you said sounds fine however.

**Comment by ExcessiveUseOfCobblestone** [comment]
should make it damage the user's arms, im sure even if you can punch through walls it probably isn't good for you to do repeatedly and haphazardly .

**Comment by Shadowflame909** [comment]
Bro..Hulk has been overnerfed.

Tasers are gone so it lost stun immunity. (since it's stunnable by disablers)

You cannot use any ranged weaponry

You're literally valid to be taken out

You cannot use any other mutation with hulk since it has 90% instability.

Now, you're going to remove the ability to delete walls?? Just remove hulk. Because there's literally nothing left. Nothing.

**Comment by necromanceranne** [comment]
Hulks are immune to stamina damage slowdown so they aren't stunned by disablers. There are also quite a lot of effects they are still immune too and also there still exist stuns in the game. Everyone in fact has one, its the shove.

You underestimate how strong hulk truly is. Also they aren't valid they are only valid if they are being griefing shitheads or they need to be arrested.

**Comment by Rohesie** [comment]
This PR is being superseded by #51398

**Comment by zxaber** [comment]
The power move would be merge both.

**Comment by ExcessiveUseOfCobblestone** [comment]
Maybe I missed something, i'm not sure how that PR fixes the issue presented in this one except instead of sometimes hitting it once to break you now always will hit it in 5.

**Comment by Qustinnus** [comment]
yeah I also think its a weird take rohesie, just because the other PR adds atom_integrity it doesnt mean the balance chances are automatically good.

**Comment by ExcessiveUseOfCobblestone** [comment]
I did miss the point, hence why I asked what that was.

**Comment by Shadowflame909** [comment]
Well to make my last comment more salty. Allowing any cheese 1 line change coder to edit how many punches it takes to break walls and r-walls (be it 100 or 1000.) versus removing it entirely is much better balance and can allow for certain control factors.

IE previously a traitor hijacking by r-walling escape would get entirely btfo'd by a hulk. Now the traitor has a guaranteed number of seconds they'd gain by stopping the hulk with R-walls.

So, ultimately, no one exactly knows what the maintainers want. But that's a common enough standard to not fuss about.

**Comment by zxaber** [comment]
The other PR doesn't really address the reason why I made this PR (though it might address why Rohesie planned to merge this PR after a 3 day period).

Unlike most other methods of removing walls, hulk does not have any meaningful resource or cost involved aside the initial search. Explosives are single-use, RCDs require either metal sheets or power/charge time (depending on flavor), thermite requires chemistry time, even mechs (whose auto attack wall smashing I also disagree with) require materials, part construction time, and then assembly time. Many of these costs can be subverted by research or help from others, but that's what makes them interesting. Meanwhile, once you have hulk, you can mass produce it for no real further cost and wall smashing has no cost whatsoever. The PKP's PR adds what i would consiter a comparably insignificant time cost when viewed aside other breaching mechanics.

I had sat on the idea to remove wall smashing for a while (I believe some of my posts even reference this on forum topics revolving around hulks). Even still, I did intend it to compromise somewhat; window and airlock breaking were left in. My intention was not to restrict the ability for hulks to break into places, but restrict the total amount of reconstruction necessary when dealing with rampant hulk.

If hulk without their mass grief ability is useless, my first question is why we should keep it at all. That said, if we *are* determined to keep hulk *and* allow hulks to smash walls, some sort of resource cost for doing so would do wonders.

**Comment by Rohesie** [comment]
My take was that I agree with removing the ability of hulks to instantly delete walls with a near-zero time cost.

And my take is that someone offered to improve the base code to make the damage done to walls less binary, instead of delete-or-not-delete we can give it higher granularity through amount of damage done.

Say hulks can still delete walls, but they need to throw 50 punches? That **_does_** address the issue, it's a significant time investment. Currently the other PR does it in 5 punches, not 50. That's fine, it's an improvement from what we currently have, and, as they say it, a level, a value we can easily tweak with a one-line change. An improvement all around.

I'm a bit conservative on balance changes and I'd like to see how this small step pans out before increasing it to huge values, or outright removing the ability, even more if this comes with a base code improvement (which is what I care the most about).

If you have different thoughts, @ExcessiveUseOfCobblestone , I defer to you, since you deal more with balance and have been around for longer.

If that doesn't explain it well, let me know what doubts you still have and I'll try to clarify.

**Comment by PKPenguin321** [comment]
Also if this PR is motivated by wallsmashing not having a resource cost then why doesn't it add one

**Comment by zxaber** [comment]
Because this PR is a removal PR, not a balance PR. There is no need for a resource cost on a removed feature, after all.

Should your PR win out (as seems to be the likelihood), and should players continue to roam the halls bringing down walls for no reason except they can, I'll probably make a new PR that adds a resource cost.

**Comment by PKPenguin321** [comment]
Well that's just lazy...

**Comment by Shadowflame909** [comment]
> Because this PR is a removal PR, not a balance PR. There is no need for a resource cost on a removed feature, after all.
> 
> Should your PR win out (as seems to be the likelihood), and should players continue to roam the halls bringing down walls for no reason except they can, I'll probably make a new PR that adds a resource cost.

okay mr elon saber. I gave you a good couple hours so I wouldn't spam your PR with comments.

But it does have a resource cost, as I've said before.

90% Instability, literally not allowing you to stack any other mutations with it unless you want to melt down.

The research of two mutations you then have to combine to obtain it, both requiring lots of human exploration or time waiting around for jokers.

Not being able to use any ranged weaponry once hulked out, and the crew being able to murder you in situations they'd otherwise detain others. Like the captain declaring all hulks valid because one broke the wall to his room down. 

Also, whilst they may be immune to stamina SLOWDOWN. They will still be stunned via stun batons and disablers. That health does tick down the same when dealing with stamina damage after all.

So, TLDR: You're trading the ROUNDSTART resources humans start off with. Like the choice of ranged weaponry and ai/crew protections, in exchange for wall busting and now wall busting alone.

**Comment by zxaber** [comment]
Opportunity costs are not resource costs. They are helpful for balance reasons, but not helpful alone.

I listed the Mech as having a very real cost (in both materials and also investment time *per mech*). It also has an opportunity cost in that you can only use one type of mech at any given time and you also cannot use *any* hand-held item at all while piloting it.

Opportunity costs do not limit the number of crew you can give this mutation to. They also do not limit individuals on their quests to destroy as many walls as possible. Just about every other mechanic to remove walls requires either obtaining more of a resource, or else dealing with a recharge time after so many walls (barring interactions with other departments).

I started with a removal PR because I did not deem wall smashing interesting enough to keep in any form. Obviously, I am disagreed with, and that's fine. Moving forward, my next attempt to balance this (should I believe it still needs such) would be a resource cost expended by the hulk per wall smashed (or possibly per wall hit). This would be a resource that can be replenished but nevertheless would force the hulk to stop after a certain number of walls before replenishment of said resource.

My issue isn't that hulks can smash a single wall (hence why I left in door and window smashings), it's that they can do so without stopping until the station has no interior walls, and that the geneticists can hand out this power to everyone for free.

**Comment by Kelenius** [comment]
> Unlike most other methods of removing walls, hulk does not have any meaningful resource or cost involved aside the initial search. Explosives are single-use, RCDs require either metal sheets or power/charge time (depending on flavor), thermite requires chemistry time, even mechs (whose auto attack wall smashing I also disagree with) require materials, part construction time, and then assembly time for each mech individually. Many of these costs can be subverted by research or help from others, but that's what makes them interesting.

Why are you completely overlooking deconstruction? The only cost of it is welding fuel, which is plentiful on the station. The tools required are also commonplace. Also, aren't walls drillable with regular drills?

**Comment by ExcessiveUseOfCobblestone** [comment]
@Rohesie that's fine, I'm willing to try it out with time investment before going to breaking ur arms.

**Comment by anconfuzedrock** [comment]
have you considered trying to restrict genetics from releasing hulk in game? Bombs can be even worse and are easier to get than hulk.

**Comment by Shadowflame909** [comment]
the alternate pr destroyed the all or nothing addendum orange man

**Comment by Rohesie** [comment]
The alternate PR is taking its time so I guess we are back to considering this PR.

Same deal as before, wait 72 hours for maintainers to ask for changes. and the author to fix the conflicts.

**Comment by ExcessiveUseOfCobblestone** [comment]
I am still for https://github.com/tgstation/tgstation/pull/51389#issuecomment-636403936

**Comment by Rohesie** [comment]
Could you do that?

**Comment by zxaber** [comment]
It's too late, 72 hours passed before Cobby's post, so you're *obligated* to merge this as is.

(I already spoke about the specifics with Cobby in discord, I'm gonna work on it probably tomorrow since it's my weekend.)

**Comment by Shadowflame909** [comment]
>so you're obligated to merge this as is

Coldest github comment ever told

**Comment by optimumtact** [comment]
i'm gonna oblige you to go jump

**Comment by Shadowflame909** [comment]
Oranges can just..change his mind on a nerf PR?

Oh! The Humanity!

**Comment by zxaber** [comment]
PR has been updated with the requested changes, and the description matches the new code. Smashing walls is still possible, but hurts your arms. Smashing machines now also hurts, but still much less than a wall.

File: code/game/turfs/closed/walls.dm [hunk 1]
```diff
@@ -150,6 +157,20 @@
...
+  *Deals damage back to the hulk.
+  *
+  *When a hulk manages to break a wall using their hulk smash, this deals back damage to the arm used.
+  *This is in its own proc just to be easily overridden by other wall types. Default allows for three
+  *smashed walls per arm. Also, we use CANT_WOUND here because wounds are random. Wounds are applied
+  *by hulk code based on arm damage and checked after an attack completes.
+  *Arguments:
+  **arg1 is the arm to deal damage to.
+ */
+
```
Comment (Rohesie) [inline code/game/turfs/closed/walls.dm:170]:
```suggestion
```
If there's a space between proc and documentation, it doesn't gets stored properly.
